src/site/content/en/blog/webassembly-threads/index.md

```diff
@@ -0,0 +1,472 @@
+
+```js
+import init, { initThreadPool /* ... */ } from './pkg/index.js';
+
+// Regular wasm-bindgen initialization.
+await init();
+
+// Thread pool initialization with the given number of threads
+// (pass `navigator.hardwareConcurrency` if you want to use all cores).
+await initThreadPool(navigator.hardwareConcurrency);
+
+// ...now you can invoke any exported functions as you normally would
+```
+
+It will create a pool of Workers and reuse them throughout the lifetime of the program. This
+initialization is similar to the `-s PTHREAD_POOL_SIZE=...` option in Emscripten explained earlier,
+and needs to happen before the main code to avoid the same deadlocks.
+
+There are few other configurations to set up before you can use multithreading, so make sure to
+check the [detailed documentation](https://github.com/GoogleChromeLabs/wasm-bindgen-rayon#usage) for
+complete setup steps.
+
+{% Aside 'caution' %} The same
+[caveats](https://github.com/GoogleChromeLabs/wasm-bindgen-rayon#caveats) about blocking the main
+thread apply here too. {% endAside %}
+
+Even the `sum_of_squares` example still needs to block the main thread to wait for the partial
+results from other threads.
+
+It might be a very short wait or a long one, depending on the complexity of iterators and number of
+available threads, but, to be on the safe side, browser engines actively prevent blocking the main
+thread altogether and code will throw an error. Instead, you should create a Worker, import the
+wasm-bindgen-generated code there, and expose its API with a library like
+[Comlink](https://github.com/GoogleChromeLabs/comlink) to the main thread.
+
+Check out
+[https://github.com/GoogleChromeLabs/wasm-bindgen-rayon/tree/main/demo](https://github.com/GoogleChromeLabs/wasm-bindgen-rayon/tree/main/demo)
+for an end-to-end demo showing:
+
+*   [Feature detection of
+    threads.](https://github.com/GoogleChromeLabs/wasm-bindgen-rayon/blob/e13485d6d64a062b890f5bb3a842b1fe609eb3c1/demo/wasm-worker.js#L27)
+*   [Building single- and multi-threaded versions of the same Rust
+    app.](https://github.com/GoogleChromeLabs/wasm-bindgen-rayon/blob/e13485d6d64a062b890f5bb3a842b1fe609eb3c1/demo/package.json#L4-L5)
+*   [Loading the JS+Wasm generated by wasm-bindgen in a
+    Worker.](https://github.com/GoogleChromeLabs/wasm-bindgen-rayon/blob/e13485d6d64a062b890f5bb3a842b1fe609eb3c1/demo/wasm-worker.js#L28-L31)
+*   [Using wasm-bindgen-rayon to initialize a thread
+    pool.](https://github.com/GoogleChromeLabs/wasm-bindgen-rayon/blob/e13485d6d64a062b890f5bb3a842b1fe609eb3c1/demo/wasm-worker.js#L32)
+*   Using Comlink to [expose Worker's
+    API](https://github.com/GoogleChromeLabs/wasm-bindgen-rayon/blob/e13485d6d64a062b890f5bb3a842b1fe609eb3c1/demo/wasm-worker.js#L44-L46)
+    to [the main
+    thread](https://github.com/GoogleChromeLabs/wasm-bindgen-rayon/blob/e13485d6d64a062b890f5bb3a842b1fe609eb3c1/demo/index.js#L25-L29).
+
+## Real-world use-cases
```

Comment:
This is the interesting part, I suggest either adding a brief summary of exciting real world use cases to the intro or just moving the entire section at the top of the post. Always try to explain "why" before "how to".

Reply:
I explained "why" in the very beginning of the post, but specific examples feels like a "nice to have" part rather than something to lead with...

Reply:
Chatted with Rachel about this, you have a point, disregard the suggestion.